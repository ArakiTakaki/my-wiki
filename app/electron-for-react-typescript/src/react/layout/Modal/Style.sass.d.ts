export const root: string;
export const content: string;
export const footerContent: string;
export const left: string;
export const right: string;
