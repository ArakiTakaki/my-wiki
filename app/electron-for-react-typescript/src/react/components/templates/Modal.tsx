import React from 'react';

interface Props {
  children?: any;
}
const Modal = (props: Props) => {
  return <h1>SampleasdfFunctionalComponent</h1>;
};

export default Modal;
