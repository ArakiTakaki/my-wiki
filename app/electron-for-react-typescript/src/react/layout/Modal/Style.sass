
.root
  position: fixed
  top: 0
  left: 0
  display: hidden
  box-sizing: border-box
  width: 100vw
  height: 100vh
  background-color: rgba(#000, 0.3)

.content
  position: absolute
  top: 50%
  left: 50%
  display: flex
  flex-direction: column
  justify-content: space-between
  min-width: 30vw
  min-height: 25vh
  max-height: 100vh
  overflow: scroll
  background-color: #fff
  border-radius: 8px
  box-shadow: 0px 4px 16px 1px rgba(#000, 0.2)
  transform: translate(-50%,-50%)

  & > header
    padding: 16px

    & > h1
      font-weight: 600
      font-size: 1.9rem
      text-indent: 0.5em

  & > section
    min-height: 20vh
    max-height: 50vh
    padding: 16px 8px
    overflow: scroll

    & > p
      margin-top: 10px
      font-size: 1.2rem
      text-indent: 0.5em

.footerContent
  display: flex
  justify-content: space-between
  font-size: 1.9rem
  text-indent: 0.5em
  border-top: 1px solid 555

  & > .left
    width: 100%
    padding: 8px 16px
    color: #FFFFFF
    font-weight: 800
    font-size: 1.2rem
    background-color: #E26964


  & > .right
    width: 100%
    padding: 8px 16px
    color: #FFFFFF
    font-weight: 800
    font-size: 1.2rem
    background-color: #4E9C95