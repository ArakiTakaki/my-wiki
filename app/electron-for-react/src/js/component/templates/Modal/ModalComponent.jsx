import React from 'react';
import PropTypes from 'prop-types';
import styles from './ModalStyle.sass';

class Modal extends React.Component {
  render() {
    //TODO write component detail
    return null;
  }
}

Modal.propTypes = {
  //TODO props: PropTypes.string.isRequired
  children: PropTypes.any
};

export default Modal;
