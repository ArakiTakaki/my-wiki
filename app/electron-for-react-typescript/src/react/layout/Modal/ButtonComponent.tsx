import React, { useCallback } from 'react';
import Style from './Style.sass';

const rippleMiddleware = (event) => {
  return event;
};

export interface IButton {
  isSingle: boolean;
  primaryText: string;
  secondaryText?: string;
  onClickPrimary?(event: React.MouseEventHandler | void): void;
  onClickSecondary?(event: React.MouseEventHandler | void): void;
}
const ButtonTypes = (props: IButton) => {
  // middleware
  const onClickPrimary = useCallback(
    (event) => {
      if (props.onClickPrimary == null) {
        return;
      }
      props.onClickPrimary(event);
    },
    [props.onClickPrimary]
  );
  const onClickSecondary = useCallback(
    (event) => {
      if (props.onClickSecondary == null) {
        return;
      }
      props.onClickSecondary(event);
    },
    [props.onClickSecondary]
  );

  return (
    <>
      {props.isSingle ? null : (
        <button className={Style.right} onClick={onClickSecondary}>
          {props.secondaryText}
        </button>
      )}
      <button className={Style.left} onClick={onClickPrimary}>
        {props.primaryText}
      </button>
    </>
  );
};

export default ButtonTypes;
