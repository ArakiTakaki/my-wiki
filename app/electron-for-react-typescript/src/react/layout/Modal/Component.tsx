import React, { ReactComponentElement } from 'react';
import Style from './Style.sass';
import { inject, observer } from 'mobx-react';
import { ModalType } from '@/react/store/ModalStore';

interface Props {
  children?: any;
  modal?: ModalType;
}
interface State {}

@inject('modal')
@observer
class Modal extends React.Component<Props, State> {
  public render() {
    const { modal } = this.props;
    if (modal == null || !modal.show) {
      return <div />;
    }

    return (
      <div className={Style.root}>
        <div className={Style.content}>
          <header>{modal.title}</header>
          <section>
            <p>よろです</p>
            <p>{modal.content}</p>
            <p>{modal.content}</p>
            <p>{modal.content}</p>
            <p>{modal.content}</p>
            <p>てすと</p>
          </section>
          <footer className={Style.footerContent}>
            <button className={Style.left}>Prev</button>

            <button className={Style.right}>Next</button>
          </footer>
        </div>
      </div>
    );
  }
}

export default Modal;
