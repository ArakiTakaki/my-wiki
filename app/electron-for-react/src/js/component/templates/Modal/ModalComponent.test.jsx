import React from 'react';
import Modal from './ModalComponent';
import renderer from 'react-test-renderer';
// react16.8で使用できないので一旦保留
// import { shallow } from 'enzyme';
// import { configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// configure({ adapter: new Adapter() });

// snapshot test
test('Modal SnapShot', () => {
  let component = null;
  component = renderer.create(<Modal />).toJSON();
  expect(component).toMatchSnapshot();
});

/*
test('Modal enzyme', () => {
  const wrapper = shallow(<Modal />);
  // TODO コンポーネントのテストを記載する
  // expect(wrapper.text()).toBe('InputType:test');
});
*/
