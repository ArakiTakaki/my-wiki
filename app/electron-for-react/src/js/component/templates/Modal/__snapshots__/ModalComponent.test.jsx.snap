// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal SnapShot 1`] = `null`;
